test/integration/connection-test.coffee

```diff
@@ -1106,3 +1105,48 @@ exports.badArithAbort = (test) ->
     connection = new Connection(config)
 
   test.done()
+
+testDateFirstImpl = (test, dateFirst) =>
+  test.expect(3)
+  config = getConfig()
+  config.options.datefirst = dateFirst
+
+  connection = new Connection(config)
+
+  request = new Request('select @@datefirst', (err) ->
+    test.ifError(err)
+    connection.close()
+  )
+
+  request.on('row', (columns) ->
+    dateFirst = columns[0].value
+    test.strictEqual(dateFirst, dateFirst || 7)
```

Comment:
Change variable to name something different than the parameter passed to the function, say datefirstReturned. As it stands, you're comparing the variable to itself.

Nit: I suggest changing parameter name from 'dateFirst' to 'datefirst' for consistency.